server/plugin_release.go

```diff
@@ -4,396 +4,699 @@
 package server
 
 import (
-	"bufio"
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
 	"context"
-	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"net/http"
 	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
 	"strings"
 	"time"
 
-	"github.com/google/go-github/github"
-	"golang.org/x/crypto/ssh"
-	"golang.org/x/oauth2"
-
-	scp "github.com/cpanato/go-scp"
-	"github.com/cpanato/go-scp/auth"
-	"github.com/eugenmayer/go-sshclient/sshwrapper"
-
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/eugenmayer/go-sshclient/sshwrapper"
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/ssh"
 )
 
-var client *github.Client
-var ctx = context.Background()
+var ErrTagExists = errors.New("tag already exists.")
+
+// cutPlugin entry point to cutting a release for a plugin.
+// This method DOES NOT generate github plugin release asset (<plugin>.tar.gz).
+// It assumes the plugin release asset to be available on the repository's release.
+// This generates:
+// 1. Plugin signature (uploaded to github)
+// 2. Platform specific plugin tars and their signatures (uploaded to s3 release bucket)
+func cutPlugin(ctx context.Context, cfg *MatterbuildConfig, client *github.Client, owner, repositoryName, tag string) error {
+	pluginAsset, err := getPluginAsset(ctx, client, owner, repositoryName, tag)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get plugin asset")
```

Comment:
Not block: `Wrapf` is not strictly needed. `errors.Wrap` is enough. Feel free to change it in the whole file if you want.

```suggestion
		return errors.Wrap(err, "failed to get plugin asset")
```